[Bug] TimeoutException encountered while accessing the admin's getMessageById API in version 3.0.7

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.


### Version

OS：centos7
Jdk：17
Pulsar version：3.0.7

### Minimal reproduce step

1. Add a test case to the testGetMessageById method in PersistentTopicsTest.java.

2. Specifically, you can add the following code:
`Assert.expectThrows(PulsarAdminException.ServerSideErrorException.class, () -> {
    admin.topics().getMessageById(topicName1, id1.getLedgerId(), id1.getEntryId() + 10);
});
`

3. Run this test case to reproduce the issue. You will encounter the following error:
`Caused by: org.apache.pulsar.client.admin.PulsarAdminException$TimeoutException: java.util.concurrent.TimeoutException
    at org.apache.pulsar.client.admin.internal.BaseResource.sync(BaseResource.java:347)
    at org.apache.pulsar.client.admin.internal.TopicsImpl.getMessageById(TopicsImpl.java:1010)
    at org.apache.pulsar.broker.admin.PersistentTopicsTest.lambda$testGetMessageById$11(PersistentTopicsTest.java:1385)
    at org.testng.Assert.expectThrows(Assert.java:2440)
    ... 29 more
`

### What did you expect to see?

The issue occurs when trying to query a non-existent message, which usually happens when a topic is newly created but hasn't received any traffic yet. In such cases, querying some information about the topic might invoke this API, leading to a timeout.

For this scenario, I would expect a fast failure, rather than being blocked until the timeout occurs.

### What did you see instead?

What I observed instead is that the getMessageById request gets blocked until the timeout occurs.

The hidden risk is that, since the timeout duration is uncertain, if the user has not configured a timeout (e.g., PulsarAdmin.builder().readTimeout(5, TimeUnit.SECONDS);) or if the timeout configuration is unreasonable, it can cause the TCP connection to enter a CLOSE_WAIT state. In extreme cases, this could potentially lead to a tcp.listenOverflow, which can affect other functionalities.

The following image shows a large number of connections in the CLOSE_WAIT state on the broker's 8080 port:
![image](https://github.com/user-attachments/assets/ac845f7e-e62f-4f77-9def-e54f303dda28)


### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java
@@ -2774,9 +2774,10 @@ public class PersistentTopicsBase extends AdminResource {
                         public void readEntryFailed(ManagedLedgerException exception,
                                                     Object ctx) {
                             if (exception instanceof ManagedLedgerException.LedgerNotExistException) {
-                                throw new RestException(Status.NOT_FOUND, "Message id not found");
+                                results.completeExceptionally(
+                                        new RestException(Status.NOT_FOUND, "Message id not found"));
                             }
-                            throw new RestException(exception);
+                            results.completeExceptionally(new RestException(exception));
                         }
 
                         @Override
@@ -2784,7 +2785,7 @@ public class PersistentTopicsBase extends AdminResource {
                             try {
                                 results.complete(generateResponseWithEntry(entry, (PersistentTopic) topic));
                             } catch (IOException exception) {
-                                throw new RestException(exception);
+                                results.completeExceptionally(new RestException(exception));
                             } finally {
                                 if (entry != null) {
                                     entry.release();

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java
@@ -1419,6 +1419,10 @@ public class PersistentTopicsTest extends MockedPulsarServiceBaseTest {
         Assert.expectThrows(PulsarAdminException.NotFoundException.class, () -> {
             admin.topics().getMessageById(topicName1, id2.getLedgerId(), id2.getEntryId());
         });
+
+        Assert.expectThrows(PulsarAdminException.ServerSideErrorException.class, () -> {
+            admin.topics().getMessageById(topicName1, id1.getLedgerId(), id1.getEntryId() + 10);
+        });
     }
 
     @Test

